src/main/java/redis/clients/jedis/Jedis.java

```diff
@@ -3953,6 +3953,10 @@ public void close() {
     }
   }
 
+  public Pool<Jedis> getDataSource() {
+    return dataSource;
+  }
```

Comment:
@Shawyeok @yangbodong22011 @dengliming I'm not so sure about this change. Can we consider not exposing or at least limiting the access of `dataSource`?

Reply:
Maybe `protected` would be better?

Reply:
Aha, I just keep it as same as setter's access modifier.
```java
  public void setDataSource(Pool<Jedis> jedisPool) {
    this.dataSource = jedisPool;
  }
```

Feel weird that setter is `public`, but getter is `protected` or `package`, maybe we should change both together?